### PR TITLE
Closes #2359 - Extend Custom Fields of Workbasket

### LINKED ISSUE #2359: Extend Custom Fields of Workbasket
Extending Custom Fields of Workbasket from 4 to 8

### PR DESCRIPTION
**Pull request by jamesrdi**
https://sonarcloud.io/summary/new_code?id=jamesrdi_taskana&branch=TSK-2359

---
Release Notes:

```

```
 
### For the submitter:
- [ ] I updated the [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview) and will supply links to the specific files
- [x] I did not update the [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview)
- [x] I included a link to the [SonarCloud branch analysis](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1019969636/SonarCloud+Integration)
- [ ] After integration of the PR, I added a description of changes on the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana)
- [x] I did not update the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana)
- [ ] I put the ticket in review
- [ ] After integration of the pull request, I verified our [bluemix test environment](http://taskana.mybluemix.net/taskana) is not broken

### Verified by the reviewer:
- [ ] Commit message format → (Closes) #<Issue Number>: Your commit message. i.e. Closes #2271: Your commit message.
- [ ] Submitter's update to [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview) is sufficient
- [ ] SonarCloud analysis meets our standards
- [ ] Update of the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana) reflects changes
- [ ] PR fulfills the ticket
- [ ] Edge cases and unwanted side effects are tested
- [ ] Readability

### CODE REVIEW

File: lib/taskana-core/src/test/java/acceptance/workbasket/query/QueryWorkbasketAccTest.java [hunk 1]
```diff
@@ -575,16 +709,21 @@ Stream<DynamicTest> should_SortQueryAsc_When_OrderingByCustomAttribute() {
...
-              WORKBASKET_SERVICE
-                  .createWorkbasketQuery()
-                  .orderByCustomAttribute(customField, ASCENDING)
-                  .list();
-
-          assertThat(results)
-              .hasSizeGreaterThan(2)
-              .extracting(w -> w.getCustomField(customField))
-              .isSortedAccordingTo(CASE_INSENSITIVE_ORDER);
+          if (customField != WorkbasketCustomField.CUSTOM_1
```
Comment (ryzheboka) [inline lib/taskana-core/src/test/java/acceptance/workbasket/query/QueryWorkbasketAccTest.java:712]: Why do you need to split logic here? Could you create Workbaskets with custom fields 5,6,7,8 in the same way as the 1,2,3,4 ?

File: lib/taskana-core/src/test/java/acceptance/workbasket/query/QueryWorkbasketAccTest.java [hunk 2]
```diff
@@ -597,17 +736,63 @@ Stream<DynamicTest> should_SortQueryDesc_When_OrderingByCustomAttribute() {
...
-              WORKBASKET_SERVICE
-                  .createWorkbasketQuery()
-                  .orderByCustomAttribute(customField, DESCENDING)
-                  .list();
-
-          assertThat(results)
-              .hasSizeGreaterThan(2)
-              .extracting(w -> w.getCustomField(customField))
-              .isSortedAccordingTo(CASE_INSENSITIVE_ORDER.reversed());
+          if (customField != WorkbasketCustomField.CUSTOM_1
```
Comment (ryzheboka) [inline lib/taskana-core/src/test/java/acceptance/workbasket/query/QueryWorkbasketAccTest.java:739]: Why do you need to split logic here? Could you create Workbaskets with custom fields 5,6,7,8 in the same way as the 1,2,3,4 ?

File: lib/taskana-core/src/test/java/acceptance/workbasket/query/QueryWorkbasketAccTest.java [hunk 3]
```diff
@@ -597,17 +736,63 @@ Stream<DynamicTest> should_SortQueryDesc_When_OrderingByCustomAttribute() {
...
+            WORKBASKET_SERVICE.deleteWorkbasket(wb1.getId());
+            WORKBASKET_SERVICE.deleteWorkbasket(wb2.getId());
+          }
+
+          checkSortCustomDsc(customField);
         };
     return DynamicTest.stream(iterator, c -> "for " + c, test);
   }
+
+  private Workbasket createDefaultWorkbasketWithCustom(
```
Comment (ryzheboka) [inline lib/taskana-core/src/test/java/acceptance/workbasket/query/QueryWorkbasketAccTest.java:758]: Can we create Workbaskets with the custom fields 5,6,7,8 in the same way as the Workbaskets with the fields 1,2,3,4? What do you think about it?

File: lib/taskana-core/src/test/java/acceptance/workbasket/query/QueryWorkbasketAccTest.java [hunk 4]
```diff
@@ -597,17 +736,63 @@ Stream<DynamicTest> should_SortQueryDesc_When_OrderingByCustomAttribute() {
...
+    workbasket.setCustomField(workbasketCustomField, value);
+    workbasket = WORKBASKET_SERVICE.createWorkbasket(workbasket);
+    WorkbasketAccessItem wbai =
+        WORKBASKET_SERVICE.newWorkbasketAccessItem(workbasket.getId(), "user-1-2");
+    wbai.setPermission(WorkbasketPermission.READ, true);
+    WORKBASKET_SERVICE.createWorkbasketAccessItem(wbai);
+    return workbasket;
+  }
+
+  private void checkSortCustomAsc(WorkbasketCustomField customField) {
```
Comment (ryzheboka) [inline lib/taskana-core/src/test/java/acceptance/workbasket/query/QueryWorkbasketAccTest.java:773]: Why do you outsource this method?

File: lib/taskana-core/src/test/java/acceptance/workbasket/query/QueryWorkbasketAccTest.java [hunk 5]
```diff
@@ -597,17 +736,63 @@ Stream<DynamicTest> should_SortQueryDesc_When_OrderingByCustomAttribute() {
...
+            .orderByCustomAttribute(customField, ASCENDING)
+            .list();
+
+    assertThat(results)
+        .hasSizeGreaterThan(2)
+        .extracting(w -> w.getCustomField(customField))
+        .isSortedAccordingTo(CASE_INSENSITIVE_ORDER);
+  }
+
+  private void checkSortCustomDsc(WorkbasketCustomField customField) {
```
Comment (ryzheboka) [inline lib/taskana-core/src/test/java/acceptance/workbasket/query/QueryWorkbasketAccTest.java:786]: Why do you outsource this method? Let's discuss this :)

**Review by ryzheboka — CHANGES_REQUESTED** [review]

File: lib/taskana-core/src/test/java/acceptance/workbasket/query/QueryWorkbasketAccTest.java [hunk 6]
```diff
@@ -455,6 +457,100 @@ void testQueryForCustom4Like() {
     assertThat(results).hasSize(5);
   }
 
+  @WithAccessId(user = "admin")
+  @Test
+  void testQueryForCustom5In() throws Exception {
```
Comment (ryzheboka) [inline lib/taskana-core/src/test/java/acceptance/workbasket/query/QueryWorkbasketAccTest.java:462]: Could you pleade rename the tests according to "should_..._When_..." ?

File: lib/taskana-core/src/test/java/acceptance/workbasket/query/QueryWorkbasketAccTest.java (hunk 3 above)
Comment (ryzheboka) [reply lib/taskana-core/src/test/java/acceptance/workbasket/query/QueryWorkbasketAccTest.java:710]: Where do you use the "createDefaultWorkbasketWithCustom" method?

File: common/taskana-common/src/main/resources/sql/postgres/taskana_schema_update_6.3.1_to_6.4.0_postgres.sql [hunk 7]
```diff
@@ -0,0 +1,12 @@
+-- this script updates the TASKANA database schema from version 6.3.1 to version 6.4.0.
+SET
+search_path = %schemaName%;
+
+INSERT INTO TASKANA_SCHEMA_VERSION (ID, VERSION, CREATED)
+VALUES (nextval('TASKANA_SCHEMA_VERSION_ID_SEQ'), '6.4.0', CURRENT_TIMESTAMP);
+
+ALTER TABLE WORKBASKET
+    ADD COLUMN COLUMN_5 VARCHAR(255) NULL,
```
Comment (ryzheboka) [inline common/taskana-common/src/main/resources/sql/postgres/taskana_schema_update_6.3.1_to_6.4.0_postgres.sql:9]: Did you mean "CUSTOM_5" etc. ? Please change this

File: common/taskana-common/src/main/resources/sql/h2/taskana_schema_update_6.3.1_to_6.4.0_h2.sql [hunk 8]
```diff
@@ -0,0 +1,13 @@
+-- this script updates the TASKANA database schema from version 6.3.1 to version 6.4.0.
+SET SCHEMA %schemaName%;
+
+INSERT INTO TASKANA_SCHEMA_VERSION (ID, VERSION, CREATED)
+VALUES (nextval('TASKANA_SCHEMA_VERSION_ID_SEQ'), '6.4.0', CURRENT_TIMESTAMP);
+
+ALTER TABLE WORKBASKET
+  ADD (
+  `COLUMN_5` VARCHAR(255) NULL,
```
Comment (ryzheboka) [inline common/taskana-common/src/main/resources/sql/h2/taskana_schema_update_6.3.1_to_6.4.0_h2.sql:9]: Did you mean "CUSTOM_5" etc?

Comment (ryzheboka) [reply common/taskana-common/src/main/resources/sql/h2/taskana_schema_update_6.3.1_to_6.4.0_h2.sql:9]: Why do you use different syntax here compared to other h2 update scripts?

**Review by ryzheboka — APPROVED** [review]
